js/js_test.go

```diff
@@ -571,3 +571,20 @@ func TestUint8Array(t *testing.T) {
 		t.Errorf("Non-empty byte array is not externalized as a Uint8Array")
 	}
 }
+
+func TestInternalizeZeroVal(t *testing.T) {
+	type S struct {
+		*js.Object
+
+		string `js:"string"`
```

Comment:
I think `Interface()` is another test case altogether because it [uses the Javascript type of the value to decide which Go type to resolve to](https://github.com/gopherjs/gopherjs/blob/3496c6f94e1b945f3fc9580f6982675c0a74cd6c/compiler/prelude/jsmapping.go#L258-L308). So in the context of this discussion, I suspect use of `Interface()` wouldn't actually be a problem per se.

What you've proposed transpiles to:

```javascript
$assertType($internalize(o.A, $emptyInterface), $Int8)
```

So we still need to have test cases for all the Go types... because it's `$internalize` that we're ultimately testing here.